from airflow import DAG
from airflow.operators.python import PythonOperator
from datetime import timedelta, datetime
import random

# ì‹¤íŒ¨ ì½œë°± í•¨ìˆ˜
def faile_callback(context):
    print("ðŸ’¥ Task Failed!")
    print("Task ID:", context['task_instance'].task_id)
    print("Dag ID:", context['dag'].dag_id)    

# ì‹¤íŒ¨í•  í™•ë¥  í•¨ìˆ˜
def unstable_fuc():
    if random.random() < 0.7:
        raise Exception("simulated failure!")
    print("success")

with DAG(
    dag_id = 'faile_task',
    start_date=datetime(2025, 4, 17),
    schedule_interval=None,
    catchup=False,
    default_args={
        'retries': 3,
        'retry_delay' : timedelta(seconds=10),
        'on_faile_callback': faile_callback
    }

) as dag:
    PythonOperator(
        task_id = 'task',
        python_callable=unstable_fuc
    )